core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1089,7 +1089,7 @@ class DAGScheduler(
         // To avoid UI cruft, ignore cases where value wasn't updated
         if (acc.name.isDefined && !updates.isZero) {
           stage.latestInfo.accumulables(id) = acc.toInfo(None, Some(acc.value))
-          event.taskInfo.accumulables += acc.toInfo(Some(updates.value), Some(acc.value))
+          event.taskInfo.accumulables ::= acc.toInfo(Some(updates.value), Some(acc.value))
```

Comment:
On the other hand, code on the `StagePage` uses `accumulables` like this:

```
    val externalAccumulableReadable = info.accumulables
      .filterNot(_.internal)
      .flatMap { a =>
        (a.name, a.update) match {
          case (Some(name), Some(update)) => Some(StringEscapeUtils.escapeHtml4(s"$name: $update"))
          case _ => None
        }
      }
```

This seems to be assuming that you'll only have one update per task.